.eslintrc.js

```diff
@@ -27,5 +25,14 @@ module.exports = {
      * https://eslint.org/docs/rules/func-names
      */
     "func-names": "off",
+
+    "max-len": ["error", 256, 2, {
+            ignoreUrls: true,
+            ignoreComments: false,
+            ignoreRegExpLiterals: true,
+            ignoreStrings: false,
+            ignoreTemplateLiterals: false,
+          }],
+    'no-plusplus': 'off'

```

Comment:
Why this change?

Reply:
The "no-plusplus" rule by default disables unary operators.  The unary operator is used several times in the pagination logic, and is a more efficient way to write the code as it reduces the number of lines needed. I didn't see why it should be omitted, so I set it to 'off'.

As for the rest, I will refactor the code to abide by the default rules and remove those conditions.

Reply:
Often these things are disabled due to bugs that don't seem obvious, but people find through experience.  You can read a bit about why they disabled it in https://github.com/airbnb/javascript/issues/540.

In modern JS, using for-loops like you're doing with a lot of `i++` isn't really necessary anymore.  It's more bug prone than using for-of, `forEach`, `map`, etc.  I'd rewrite your code to not need to use an index counter at all.  That's the point of this rule.

Reply:
Okay fair enough I'll make the changes.